### PR TITLE
doc/rst2html: some few fixes for enumerated and bullet lists

### LINKED ISSUE #16041: rst2html list numbering wrong
rst2html list numbering wrong, causing numbering issues (eg see https://github.com/nim-lang/Nim/pull/16042)

### Example
```nim

1. Changes should not introduce any trailing whitespace.
   Always check your changes for whitespace errors using ``git diff --check``
   or add the following ``pre-commit`` hook:

   .. code-block:: sh

      #!/bin/sh
      git diff --check --cached || exit $?
2. Changes should not introduce any trailing whitespace.

   Always check your changes for whitespace errors using ``git diff --check``
   or add the following ``pre-commit`` hook:

   .. code-block:: sh

      #!/bin/sh
      git diff --check --cached || exit $?

3. Changes should not introduce any trailing whitespace.

   Always check your changes for whitespace errors using ``git diff --check``
   or add the following ``pre-commit`` hook:

   .. code-block:: sh

      #!/bin/sh
      git diff --check --cached || exit $?
4. Changes should not introduce any trailing whitespace.
   Always check your changes for whitespace errors using ``git diff --check``
   or add the following ``pre-commit`` hook:

   .. code-block:: sh

      #!/bin/sh
      git diff --check --cached || exit $?

5. Describe your commit and use your common sense.
   Example commit message:

6. Describe your commit and use your common sense.
   Example commit message:

7. Describe your commit and use your common sense.
   Example commit message:


8. Describe your commit and use your common sense.
   Example commit message:

```

### Current Output

image

### Expected Output
* the numbering should be honored instead of 1,2,3,1,1,2,3,1
* I'm not sure the formatting of item 3 is correct

### Additional Information
* devel 1.5.1 acf8316e502cf46358df220ef4250b7ca96c6c64
* see `nimdoc/rsttester.nim` to run rst tests
* ideally nimdoc/rsttester.nim would be refactored with nimdoc/tester.nim to avoid code duplication

### PR DESCRIPTION
**Pull request by a-mr**
First of all, fix #16041 : enumerated lists parsing is re-implemented — now there is no requirement, that it should contain some line after enumerator line, and it does not produce spurious block quotes as in #16041 . Also `(1)` and `(2)` were not parsed as enumerated list in `nimdoc/rst2html/expected/rst_examples.html`, most probably it was the same problem.

Also now rst.nim correctly handles alphabetic characters as enumerators:
```
a. string1
b. string2
```
while previously it turned them into `1. string1 2. string2`.

According to the spec rst.nim checks that
* enumerator numbers/letters are in sequence and produces separate lists if they are not
* items belong to the same kind, and otherwise parses as 2 separate lists:
   ```
   1. list1
   a. list2
   ```

It can now generate auto-enumerator properly:
```
2. expect2
#. expect3

b. expect b
#. expect c
```

The first commit addresses a few blank lines problem for enumerated and bullet lists:

```rst
* this is c1
* this is c2

* this is c3


* this is c4



* this is c5
```
was not parsed as a single list, while [Multiple successive blank lines are equivalent to a single blank line](https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#id41)

cc @narimiran

### CODE REVIEW

**Comment by timotheecour** [comment]
* great, also seems to fix `nim doc`; maybe add a doc comment in the test file `nimdoc/testproject/testproject.nim` that exercises this?

* this should be documented somewhere after (or in this) PR, maybe manual.rst or doc/docgen.rst

**Comment by a-mr** [comment]
@narimiran please review.

File: tests/stdlib/trstgen.nim [hunk 1]
```diff
@@ -361,6 +361,144 @@ Test1
...
+
+    # check that nested enumerated lists work
+    let input3 = dedent """
+      1.  a) string1
+      2. string2
+      """
+    let output3 = rstToHtml(input3, {roSupportMarkdown}, defaultConfig())
+    assert count(output3, "<ol ") == 2
+    assert count(output3, "</ol>") == 2
+    assert "<li>string1</li>" in output3 and "<li>string2</li>" in output3
```
Comment (timotheecour) [inline tests/stdlib/trstgen.nim:418]: tests use doAssert or check, not assert (otherwise noop with --assertions:off)

Comment (narimiran) [reply tests/stdlib/trstgen.nim:418]: Not important.

**Comment by narimiran** [comment]
Looks good!

Since we also support markdown, the following is a valid ordered list:
```
1. fist
1. second
1. third
```
(The behaviour is the same as in your `#.` examples: auto-increment)

Could you add support for that too?

**Comment by Araq** [comment]
> Could you add support for that too?

Can be done in a followup PR, I'm merging this now.

**Pull request merged**